semantics/src/namespace/types.rs

```diff
@@ -133,13 +139,34 @@ pub struct FeString {
     pub max_size: usize,
 }
 
+impl Integer {
+    pub fn is_signed(&self) -> bool {
+        match self {
+            Integer::I256 => true,
+            Integer::I128 => true,
+            Integer::I64 => true,
+            Integer::I32 => true,
+            Integer::I16 => true,
+            Integer::I8 => true,
```

Comment:
Having written :point_up: I could also see the case for splitting signed and unsigned integers into their own enums but it may lead to less nice code elsewhere...so, I'm fine to leave it as is but I would also be open to change it.